### PR TITLE
Mainfest Structure: Add documentation

### PR DESCRIPTION
**Pull request by DunderRoffe**
Add documentation for the manifest structure including what
requirements it is developed to fulfill and how it works.

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

### CODE REVIEW

File: manifest_structure.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+---------
+
+![IMAGE OF MANIFEST STRUCTURE](manifest_structure.svg)
+
+In the above image, `Base` is the manifest file containing mappings to the git
+repositories which are required by every PELUX build (i.e. `PELUX-base.xml`).
+`Qt` is the manifest file that contain mappings to the git repositories required
+in every PELUX QtAS build which is not already covered in `Base`.
+
+In contrast to `Base` and `Qt` there multiple instances of `Bsp` and
```
Comment (breakreturn) [inline manifest_structure.md]: "there are" ?

File: manifest_structure.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+`Bsp + QtAs`. However, each such instance holds mappings to git repositories
+required by specific hardware target. This means that there is no one "Bsp"
+or "Bsp + QtAS" manifest file, but one file each per hardware target. An example
+of this are `pelux-intel.xml` and `pelux-intel-qt.xml`this which represent
+specific implementations of `Bsp` and `Bsp + QtAS` for the intel hardware target.
+
+Using this this structure minimizes the number of manifest files in the
+repository. The amount of files can be calculated by the formula `2 * n + 2`
```
Comment (breakreturn) [inline manifest_structure.md]: Why is it relevant to describe a formula for calculating this? I think it adds verbosity and not much use.

Comment (DunderRoffe) [inline manifest_structure.md]: The thought was to explain how the solution scales. But it might be that I over analyzed this and the scaling is trivial?

Comment (breakreturn) [inline manifest_structure.md]: IMHO it's enough to state that the purpose is to reduce amount of files etc. (as you already do).

File: doc/manifest_structure.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+![Image of manifest structure](manifest_structure.svg)
+
+In the above image, `Base` is the manifest file containing mappings to the git
+repositories which are required by every PELUX build (i.e. `PELUX-base.xml`).
+`Qt` is the manifest file that contain mappings to the git repositories required
+in every PELUX QtAS build which is not already covered in `Base`.
+
+In contrast to `Base` and `Qt` there are multiple instances of `Bsp` and
+`Bsp + QtAs`. However, each such instance holds mappings to git repositories
```
Comment (GordanM) [inline doc/manifest_structure.md]: Shouldn't "Bsp" be in all caps (BSP)?

Comment (DunderRoffe) [inline doc/manifest_structure.md]: Yes, you are correct. "QtAs" should probably be "QtAS" as well.
Will fix.

Comment (GordanM) [inline doc/manifest_structure.md]: Thanks!

**Pull request merged**